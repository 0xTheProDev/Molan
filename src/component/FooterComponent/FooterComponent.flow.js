// @flow
export type FooterComponentProps = {
  ServiceStatus: number,
  reloadServiceStatus: () => void,
};

export type CopyRightHolderItem = {
  title: string,
  url: string,
};

export type ServiceStatus = {
  icon: string,
  color: string,
  label: string,
};

export type FooterComponentInternalProps = {
  CopyRightTitleText: string,
  CopyRightHolders: Array<CopyRightHolderItem>,
  ServiceStatusTitleText: string,
  ServiceStatus: ServiceStatus,
  reloadServiceStatus: () => void,
};
