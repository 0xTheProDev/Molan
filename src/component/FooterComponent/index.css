.footer-content {
    height: 60px !important;
    background-color: #FFF8DC !important;
}

.service-status {
    border: 1px solid rgba(34,36,38,.15) !important;
    border-radius: 10px !important;
    cursor: pointer !important;
}

.service-status:hover {
    border-color: #000 !important;
}

.service-status:active, .service-status:focus {
    border-color: #85b7d9 !important;
}

.copyright {
    float: left;
}

.footer-cols {
    float: right;
}

@media (min-width: 600px) {
    .service-status {
        padding: 5px 0 !important;
        margin: 0 180px !important;
    }

    .footer-cols {
        width: 40%;
    }
}

@media (max-width: 599px) {
    .service-status {
        padding: 5px !important;
    }
}