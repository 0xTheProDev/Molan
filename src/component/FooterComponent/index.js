import React, { Component } from "react";
import { Grid, Segment, Divider } from "semantic-ui-react";
import "./index.css";

export default class FooterComponent extends Component {
    render() {
        return (
            <div className="footer-content">
                <div className="copyright">
                    <Segment basic>
                        &copy; All rights reserved by <a href="https://github.com/Tech-Mantra" target="_blank" rel="noopener noreferrer">Tech-Mantra</a>
                    </Segment>
                </div>
                <div className="footer-cols">
                    <Grid columns={3} relaxed>
                        <Grid.Column>
                          <Segment basic>About</Segment>
                        </Grid.Column>
                        <Divider vertical hidden/>
                        <Grid.Column>
                          <Segment basic>Privacy Policy</Segment>
                        </Grid.Column>
                        <Divider vertical hidden/>
                        <Grid.Column>
                          <Segment basic>Terms of Usage</Segment>
                        </Grid.Column>
                    </Grid>
                </div>
            </div>
        );
    }
}
