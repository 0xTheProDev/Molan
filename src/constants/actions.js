export const ACTIONS = {

  /** To Get the Health Status of Runner Application */
  GET_STATUS: 'GET_STATUS',

  /** To Submit Code and Language Action */
  SUBMIT_CODE: 'SUBMIT_CODE',
};

export default ACTIONS;
