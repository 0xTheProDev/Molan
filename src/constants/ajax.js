/** Base URL to Runner Application (e.g., `localhost:5001`) */
export const BASE_URL = 'http://localhost:5001/molan-ide/us-central1/default-app';

/** HTTP Methods */
export const METHODS = Object.freeze({
  GET:      'GET',
  POST:     'POST',
  PUT:      'PUT',
  DELETE:   'DELETE',
});

/** Allowed HTTP Methods for Ajax */
export const ALLOWED_METHODS = Object.freeze([
  METHODS.GET,
  METHODS.POST,
]);

/** Error string for Invalid Method */
export const METHOD_NOT_ALLOWED = 'Method not allowed for Ajax Request';
