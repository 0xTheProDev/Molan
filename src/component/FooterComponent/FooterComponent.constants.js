// @flow
import { SERVICE_STATUS } from 'constants/services';
import { CopyRightHolderItem, ServiceStatus } from './FooterComponent.flow';

export const COPYRIGHT_HOLDERS: Array<CopyRightHolderItem> = Object.freeze([
  {
    title: 'Bytes Club',
    url: 'https://github.com/BytesClub',
  },
  {
    title: 'Progyan Bhattacharya',
    url: 'https://progyan1997.github.io',
  }
]);


const SERVICE_STATUS_LOADING: ServiceStatus = Object.freeze({
  icon: 'question',
  color: 'yellow',
  label: 'Loading',
});

const SERVICE_STATUS_READY: ServiceStatus = Object.freeze({
  icon: 'check',
  color: 'green',
  label: 'Ok',
});

const SERVICE_STATUS_ERROR: ServiceStatus = Object.freeze({
  icon: 'close',
  color: 'red',
  label: 'Failed',
});

export const SERVICE_STATUS_OPTIONS = Object.freeze({
  LOADING: SERVICE_STATUS_LOADING,
  READY: SERVICE_STATUS_READY,
  ERROR: SERVICE_STATUS_ERROR,
});

export const SERVICE_STATUS_MAPPING = Object.freeze({
  [SERVICE_STATUS.LOADING]: SERVICE_STATUS_LOADING,
  [SERVICE_STATUS.READY]: SERVICE_STATUS_READY,
  [SERVICE_STATUS.ERROR]: SERVICE_STATUS_ERROR,
});
