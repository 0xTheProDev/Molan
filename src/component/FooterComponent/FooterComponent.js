// @flow
import React from 'react';
import { Button, Grid, Icon, Segment } from 'semantic-ui-react';
import { noop } from 'util/common';

import { FooterComponentInternalProps } from './FooterComponent.flow';
import { COPYRIGHT_HOLDERS, SERVICE_STATUS_OPTIONS } from './FooterComponent.constants';
import Styles from './FooterComponent.module.css';

export function CopyRightSection({ CopyRightTitleText, CopyRightHolders }: FooterComponentInternalProps) {
  return (
    <div className={Styles['copyright']} role="contentinfo">
      <Segment basic>
        <i>&copy;</i>
        <span className="lg-only">{CopyRightTitleText}</span>
        {
          CopyRightHolders.map((copyRightHolder) =>
            <a key={copyRightHolder.url} href={copyRightHolder.url} target="_blank" rel="noopener noreferrer">
              {copyRightHolder.title}
            </a>
          )
        }
      </Segment>
    </div>
  );
}

export function ServiceStatusSection({ ServiceStatus, ServiceStatusTitleText, reloadServiceStatus }: FooterComponentInternalProps) {
  return (
    <Grid columns={1} centered relaxed>
      <Grid.Column>
        <Segment basic textAlign="center">
          <Button className={Styles['service-status']} onClick={reloadServiceStatus} role="status">
            <Icon
              name={ServiceStatus.icon}
              color={ServiceStatus.color}
              aria-label={ServiceStatus.label}
              aria-hidden={ServiceStatus.label}
            />
            {ServiceStatusTitleText}
          </Button>
        </Segment>
      </Grid.Column>
    </Grid>
  );
}

export function FooterComponent(props: FooterComponentInternalProps) {
  return (
    <div className={Styles['footer-content']}>
      <CopyRightSection {...props} />
      <div className={Styles['footer-cols']}>
        <ServiceStatusSection {...props} />
      </div>
    </div>
  );
}

FooterComponent.defaultProps = {
  CopyRightTitleText: 'All rights reserved by',
  CopyRightHolders: COPYRIGHT_HOLDERS,
  ServiceStatusTitleText: 'Service Status',
  ServiceStatus: SERVICE_STATUS_OPTIONS.LOADING,
  reloadServiceStatus: noop,
};
