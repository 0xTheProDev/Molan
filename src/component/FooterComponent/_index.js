// @flow
import React from 'react';

import { FooterComponentProps } from './FooterComponent.flow';
import { FooterComponent } from './FooterComponent';
import { SERVICE_STATUS_MAPPING } from './FooterComponent.constants';

export default function FooterComponentFactory({ ServiceStatus, reloadServiceStatus }: FooterComponentProps) {
  const ServiceStatusOption = SERVICE_STATUS_MAPPING[ServiceStatus];

  return (
    <FooterComponent
      ServiceStatus={ServiceStatusOption}
      reloadServiceStatus={reloadServiceStatus}
    />
  );
}

export { FooterComponentProps, FooterComponentFactory as FooterComponent };
