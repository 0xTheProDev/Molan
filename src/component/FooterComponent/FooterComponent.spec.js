import React from 'react';
import { Button } from 'semantic-ui-react';

import { FooterComponent, CopyRightSection, ServiceStatusSection } from './FooterComponent';
import { SERVICE_STATUS_OPTIONS } from './FooterComponent.constants';

describe('Component::FooterComponent::FooterComponent', () => {
  it('should render with base structure', () => {
    const wrapper = shallow(<FooterComponent />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render status icon and text properly when service loaded', () => {
    const wrapper = shallow(<FooterComponent ServiceStatus={SERVICE_STATUS_OPTIONS.READY} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render status icon and text properly when service failed', () => {
    const wrapper = shallow(<FooterComponent ServiceStatus={SERVICE_STATUS_OPTIONS.ERROR} />);
    expect(wrapper).toMatchSnapshot();
  });

  describe('Test CopyRightSection', () => {
    it('should render with base structure', () => {
      const wrapper = shallow(<CopyRightSection {...FooterComponent.defaultProps} />);
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('Test ServiceStatusSection', () => {
    it('should render with base structure', () => {
      const wrapper = shallow(<ServiceStatusSection {...FooterComponent.defaultProps} />);
      expect(wrapper).toMatchSnapshot();
    });
  
    it('should render status icon and text properly when service loaded', () => {
      const wrapper = shallow(<ServiceStatusSection {...FooterComponent.defaultProps} ServiceStatus={SERVICE_STATUS_OPTIONS.READY} />);
      expect(wrapper).toMatchSnapshot();
    });
  
    it('should render status icon and text properly when service failed', () => {
      const wrapper = shallow(<ServiceStatusSection {...FooterComponent.defaultProps} ServiceStatus={SERVICE_STATUS_OPTIONS.ERROR} />);
      expect(wrapper).toMatchSnapshot();
    });

    it('should reload service status when clicked', () => {
      const reloadServiceStatusStub = jest.fn();

      const wrapper = shallow(<ServiceStatusSection {...FooterComponent.defaultProps} reloadServiceStatus={reloadServiceStatusStub} />);
      wrapper
        .find(Button)
        .simulate('click');

      expect(reloadServiceStatusStub).toHaveBeenCalled();
    });
  });
});
