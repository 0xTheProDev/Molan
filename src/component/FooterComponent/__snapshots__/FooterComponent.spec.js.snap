// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component::FooterComponent::FooterComponent Test CopyRightSection should render with base structure 1`] = `
<div
  className="copyright"
  role="contentinfo"
>
  <Segment
    basic={true}
  >
    <i>
      Â©
    </i>
    <span
      className="lg-only"
    >
      All rights reserved by
    </span>
    <a
      href="https://github.com/BytesClub"
      key="https://github.com/BytesClub"
      rel="noopener noreferrer"
      target="_blank"
    >
      Bytes Club
    </a>
    <a
      href="https://progyan1997.github.io"
      key="https://progyan1997.github.io"
      rel="noopener noreferrer"
      target="_blank"
    >
      Progyan Bhattacharya
    </a>
  </Segment>
</div>
`;

exports[`Component::FooterComponent::FooterComponent Test ServiceStatusSection should render status icon and text properly when service failed 1`] = `
<Grid
  centered={true}
  columns={1}
  relaxed={true}
>
  <GridColumn>
    <Segment
      basic={true}
      textAlign="center"
    >
      <Button
        as="button"
        className="service-status"
        onClick={[Function]}
        role="status"
      >
        <Icon
          aria-hidden="Failed"
          aria-label="Failed"
          as="i"
          color="red"
          name="close"
        />
        Service Status
      </Button>
    </Segment>
  </GridColumn>
</Grid>
`;

exports[`Component::FooterComponent::FooterComponent Test ServiceStatusSection should render status icon and text properly when service loaded 1`] = `
<Grid
  centered={true}
  columns={1}
  relaxed={true}
>
  <GridColumn>
    <Segment
      basic={true}
      textAlign="center"
    >
      <Button
        as="button"
        className="service-status"
        onClick={[Function]}
        role="status"
      >
        <Icon
          aria-hidden="Ok"
          aria-label="Ok"
          as="i"
          color="green"
          name="check"
        />
        Service Status
      </Button>
    </Segment>
  </GridColumn>
</Grid>
`;

exports[`Component::FooterComponent::FooterComponent Test ServiceStatusSection should render with base structure 1`] = `
<Grid
  centered={true}
  columns={1}
  relaxed={true}
>
  <GridColumn>
    <Segment
      basic={true}
      textAlign="center"
    >
      <Button
        as="button"
        className="service-status"
        onClick={[Function]}
        role="status"
      >
        <Icon
          aria-hidden="Loading"
          aria-label="Loading"
          as="i"
          color="yellow"
          name="question"
        />
        Service Status
      </Button>
    </Segment>
  </GridColumn>
</Grid>
`;

exports[`Component::FooterComponent::FooterComponent should render status icon and text properly when service failed 1`] = `
<div
  className="footer-content"
>
  <CopyRightSection
    CopyRightHolders={
      Array [
        Object {
          "title": "Bytes Club",
          "url": "https://github.com/BytesClub",
        },
        Object {
          "title": "Progyan Bhattacharya",
          "url": "https://progyan1997.github.io",
        },
      ]
    }
    CopyRightTitleText="All rights reserved by"
    ServiceStatus={
      Object {
        "color": "red",
        "icon": "close",
        "label": "Failed",
      }
    }
    ServiceStatusTitleText="Service Status"
    reloadServiceStatus={[Function]}
  />
  <div
    className="footer-cols"
  >
    <ServiceStatusSection
      CopyRightHolders={
        Array [
          Object {
            "title": "Bytes Club",
            "url": "https://github.com/BytesClub",
          },
          Object {
            "title": "Progyan Bhattacharya",
            "url": "https://progyan1997.github.io",
          },
        ]
      }
      CopyRightTitleText="All rights reserved by"
      ServiceStatus={
        Object {
          "color": "red",
          "icon": "close",
          "label": "Failed",
        }
      }
      ServiceStatusTitleText="Service Status"
      reloadServiceStatus={[Function]}
    />
  </div>
</div>
`;

exports[`Component::FooterComponent::FooterComponent should render status icon and text properly when service loaded 1`] = `
<div
  className="footer-content"
>
  <CopyRightSection
    CopyRightHolders={
      Array [
        Object {
          "title": "Bytes Club",
          "url": "https://github.com/BytesClub",
        },
        Object {
          "title": "Progyan Bhattacharya",
          "url": "https://progyan1997.github.io",
        },
      ]
    }
    CopyRightTitleText="All rights reserved by"
    ServiceStatus={
      Object {
        "color": "green",
        "icon": "check",
        "label": "Ok",
      }
    }
    ServiceStatusTitleText="Service Status"
    reloadServiceStatus={[Function]}
  />
  <div
    className="footer-cols"
  >
    <ServiceStatusSection
      CopyRightHolders={
        Array [
          Object {
            "title": "Bytes Club",
            "url": "https://github.com/BytesClub",
          },
          Object {
            "title": "Progyan Bhattacharya",
            "url": "https://progyan1997.github.io",
          },
        ]
      }
      CopyRightTitleText="All rights reserved by"
      ServiceStatus={
        Object {
          "color": "green",
          "icon": "check",
          "label": "Ok",
        }
      }
      ServiceStatusTitleText="Service Status"
      reloadServiceStatus={[Function]}
    />
  </div>
</div>
`;

exports[`Component::FooterComponent::FooterComponent should render with base structure 1`] = `
<div
  className="footer-content"
>
  <CopyRightSection
    CopyRightHolders={
      Array [
        Object {
          "title": "Bytes Club",
          "url": "https://github.com/BytesClub",
        },
        Object {
          "title": "Progyan Bhattacharya",
          "url": "https://progyan1997.github.io",
        },
      ]
    }
    CopyRightTitleText="All rights reserved by"
    ServiceStatus={
      Object {
        "color": "yellow",
        "icon": "question",
        "label": "Loading",
      }
    }
    ServiceStatusTitleText="Service Status"
    reloadServiceStatus={[Function]}
  />
  <div
    className="footer-cols"
  >
    <ServiceStatusSection
      CopyRightHolders={
        Array [
          Object {
            "title": "Bytes Club",
            "url": "https://github.com/BytesClub",
          },
          Object {
            "title": "Progyan Bhattacharya",
            "url": "https://progyan1997.github.io",
          },
        ]
      }
      CopyRightTitleText="All rights reserved by"
      ServiceStatus={
        Object {
          "color": "yellow",
          "icon": "question",
          "label": "Loading",
        }
      }
      ServiceStatusTitleText="Service Status"
      reloadServiceStatus={[Function]}
    />
  </div>
</div>
`;
